mission_control/src/mission_control.cpp

```diff
@@ -368,58 +273,6 @@ void MissionControlNode::reportFaultCallback(const health_monitor::ReportFault::
   }
 }
 
-bool MissionControlNode::endsWith(const std::string& str, const char* suffix)
-{
-  unsigned suffixLen = std::string::traits_type::length(suffix);
-  return str.size() >= suffixLen &&
-         0 == str.compare(str.size() - suffixLen, suffixLen, suffix, suffixLen);
-}
-
-bool MissionControlNode::FoundMissionFile()
-{
-  ROS_INFO("Mission path is [%s]", mission_path.c_str());
-
-  bool retval = false;
-  DIR* d;
-  struct dirent* dir;
-  d = opendir(mission_path.c_str());
-
-  if (d)  // if the directory is valid
-  {
-    if (endsWith(mission_path, ".xml")) return true;
-
-    while (retval == false)
-    {
-      while ((dir = readdir(d)) != NULL)  // read file in directory
-      {
-        // Condition to check regular file.
-        if (dir->d_type == DT_REG)
-        {
-          mission_path.append("/");
-          mission_path.append(dir->d_name);
-
-          ROS_INFO("Found mission file. full path is: %s\n", mission_path.c_str());
-          retval = true;
-          break;
-        }
-      }
-
-      if (retval == false)
-      {
-        ROS_INFO("sleeping");
-        usleep(250000);
-      }
-    }
-  }
-  else
-  {
-    ROS_ERROR("The directory path to the mission file is invalid: %s", mission_path.c_str());
-  }
-
-  closedir(d);
-  return retval;
-}
-
 int main(int argc, char** argv)
```

Comment:
@patapufate consider separating the `MissionControlNode` class from the executable itself. You won't be able to write gtests otherwise.

Reply:
Done